File: replace_currency.py
import os
import re

def process_file(filepath):
    with open(filepath, 'r') as f:
        content = f.read()

    original = content

    # 1. Price ($) -> Price (₹)
    content = content.replace("($)", "(₹)")
    content = content.replace("Cost ($)", "Cost (₹)")
    
    # 2. JSX text: >${ -> >₹{
    content = re.sub(r'>\$(\{)', r'>₹\1', content)
    
    # 3. Text with $: "Raw: $" -> "Raw: ₹"
    content = content.replace("Raw: $", "Raw: ₹")
    content = content.replace("Fixed: $", "Fixed: ₹")
    content = content.replace("Discount: $", "Discount: ₹")
    
    # 4. Template literals with currency: `$${ -> `₹${
    content = content.replace("`$$", "`₹$")
    # also standard string concat: "$ " -> "₹ " (maybe not safe)
    
    # 5. JSX text like > $10 or >$10
    content = re.sub(r'>\s*\$(\d)', r'> ₹\1', content)
    
    # 6. Fallbacks for strings like "$0.00" inside JSX
    content = re.sub(r'"\$(\d)', r'"₹\1', content)
    content = re.sub(r"'\$(\d)", r"'₹\1", content)
    
    if content != original:
        with open(filepath, 'w') as f:
            f.write(content)
        print(f"Updated {filepath}")

for root, dirs, files in os.walk('src'):
    for file in files:
        if file.endswith('.ts') or file.endswith('.tsx'):
            process_file(os.path.join(root, file))

File: replace_currency.js
const fs = require('fs');
const path = require('path');

function walk(dir) {
  let results = [];
  const list = fs.readdirSync(dir);
  list.forEach(file => {
    file = path.resolve(dir, file);
    const stat = fs.statSync(file);
    if (stat && stat.isDirectory()) {
      results = results.concat(walk(file));
    } else {
      if (file.endsWith('.tsx') || file.endsWith('.ts')) {
        results.push(file);
      }
    }
  });
  return results;
}

const files = walk('./src');

files.forEach(file => {
  let content = fs.readFileSync(file, 'utf8');
  const original = content;
  
  // 1. Replace literal `$` before `{` in JSX (e.g. >${value} or " > ${value}")
  // We want to replace $ only when it's text in JSX, which usually follows >, or space, or ".
  content = content.replace(/>\s*\$\{/g, '>₹{');
  content = content.replace(/"\$\{/g, '"₹{');
  
  // 2. Replace Price ($) etc.
  content = content.replace(/Price \(\$\)/g, 'Price (₹)');
  content = content.replace(/Cost \(\$\)/g, 'Cost (₹)');
  
  // 3. Replace literal $ followed by a number
  content = content.replace(/\$([0-9])/g, '₹$1');
  
  // 4. Replace specific strings
  content = content.replace(/Total: \$/g, 'Total: ₹');
  content = content.replace(/Discount: \$/g, 'Discount: ₹');
  content = content.replace(/Raw: \$/g, 'Raw: ₹');
  content = content.replace(/Fixed: \$/g, 'Fixed: ₹');
  
  // 5. Replace `\${` inside backticks where the user intended a dollar sign followed by interpolation.
  // Wait, in JS template literals, `${var}` is interpolation. A literal dollar before it is `$${var}`.
  content = content.replace(/\$\$\{/g, '₹${');
  
  // 6. In some cases, we might have `\${` in string literals? 
  // Let's just fix the remaining >${ occurrences which might not have been caught
  content = content.replace(/>\$\{/g, '>₹{');

  if (content !== original) {
    fs.writeFileSync(file, content);
    console.log('Updated ' + file);
  }
});
